server/manager/src/main/java/org/apache/accumulo/manager/TabletGroupWatcher.java

```diff
@@ -635,31 +637,34 @@ private void deleteTablets(MergeInfo info) throws AccumuloException {
       ServerColumnFamily.TIME_COLUMN.fetch(scanner);
       scanner.fetchColumnFamily(DataFileColumnFamily.NAME);
       scanner.fetchColumnFamily(CurrentLocationColumnFamily.NAME);
-      Set<String> datafiles = new TreeSet<>();
+      Set<Reference> datafilesAndDirs = new TreeSet<>();
```

Comment:
I think Reference and maybe it's subclasses are going to need a Comparator or implement Comparable to maintain the same order. It may be useful to also have them implement `equals` and `hashCode` too.

Reply:
I wasn't sure which part to compare though. I am thinking that we want to compare the whole metadataEntry string...

Reply:
I implemented the methods in 60b60de. I don't think I need a comparator since I am using the metadata string.

Reply:
I think that a typical implementation of `hashCode`, `equals`, and `compareTo` would use all of the member variables, not just one of them. Looking at `Reference`, `RelativeTabletDirectory`, and `TabletDirectory`, I think I'm a little confused as to how they are used and why. I thought that all file references in the metadata table were absolute. Given my confusion, I can't say whether or not the changes in 60b60de are sufficient. I think the changes are sufficient to order the TreeSet, I'm just not sure what the side effects are of only using one of the member variables in the comparison methods.

Reply:
> I thought that all file references in the metadata table were absolute

If it is to a file, then yes, it should be... eventually. The GC uses references to directories, and passes them around as Strings. See https://github.com/apache/accumulo/blob/76aaba0ec4e086ab535a1e88ba6ecb0f62481b2c/server/gc/src/main/java/org/apache/accumulo/gc/GarbageCollectionAlgorithm.java#L71

Reply:
Also checkout @keith-turner 's changes to prefix all tablet files with a new special encoded volume #1389

Reply:
> Also checkout @keith-turner 's changes to prefix all tablet files with a new special encoded volume https://github.com/apache/accumulo/pull/1389

Interesting, I wasn't aware of this change. I'll have to look at this further and digest it.